src/java.management.rmi/share/classes/javax/management/remote/rmi/RMIConnectionImpl.java

```diff
@@ -1578,32 +1471,24 @@ private <T> T unwrap(final MarshalledObject<?> mo,
             return null;
         }
         try {
-            final ClassLoader old = AccessController.doPrivileged(new SetCcl(cl));
+            ClassLoader old = setCcl(cl);
             try {
                 if (subject != null) {
-                    return Subject.doAs(subject, (PrivilegedExceptionAction<T>) () -> wrappedClass.cast(mo.get()));
+                    return Subject.callAs(subject, () -> wrappedClass.cast(mo.get()));
                 } else {
                     return wrappedClass.cast(mo.get());
                 }
             } finally {
-                AccessController.doPrivileged(new SetCcl(old));
+                setCcl(old);
             }
-        } catch (PrivilegedActionException pe) {
-            Exception e = extractException(pe);
+        } catch (Exception e) {
```

Comment:
Need to handle CompletionException (or maybe at Subject.callAs call)?

Reply:
link to https://github.com/openjdk/jdk/pull/22315